test/SlowTests/Issues/RavenDB-19466.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using SlowTests.Core.Utils.Entities;
+using SlowTests.Server.Documents.ETL;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues;
+
+public class RavenDB_19466:EtlTestBase 
+{
+    public RavenDB_19466(ITestOutputHelper output) : base(output)
+    {
+    }
+
+    [Fact]
+    public void CanModifyDocumentMetadataUsingEtlTransformScript()
+    {
+        using (var src = GetDocumentStore())
+        using (var dest = GetDocumentStore())
+        {
+            AddEtl(src, dest, "Users", script: @"var metadata = getMetadata(this);
```

Comment:
does this pass if you add prevent deletions in the script?
```
function deleteDocumentsBehavior(docId) {
    return false; // don't send any deletes to the other side
}

```

Reply:
I see there is a problem with the delete behavior.
If there's a call to function that sends `this` as an argument inside the script, it'll fail during the `Initialize` call [here](https://github.com/ravendb/ravendb/blob/d8d313e88d3107c21d1c489f85919d735c5fda97/src/Raven.Server/Documents/ETL/EtlTransformer.cs#L49)

Working on it

Reply:
Added the behavior function to the test.
At first, it caused errors during the test run of the script inside the `Initialize` method. 
The `getMetadata` had an additional check that ensured that the type of object given as an argument is a `BlittableObjectInstance`. 

On the script 'test run' the argument coming to the `getMetadata` wasn't a BOI, but a simple Jint `GlobalObject` so the whole ETL initialization failed.

Fixed it by an additional check inside `getMetadata` before throwing an error.